api/prometheus/v1/api_fake.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package v1
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/common/model"
+)
+
+type FakeAPI struct {
```

Comment:
If we follow this path we absolutely need to ensure this struct is not disconnected from API interface.

```suggestion
var _ API = &FakeAPI{}

type FakeAPI struct {
```